src/SegmentedString.chpl

```diff
@@ -1112,6 +1117,59 @@ module SegmentedString {
       return ranks;
     }
 
+    proc getFixesHelper(n: int, kind: Fixes, proper: bool) {
+      const lengths = getLengths() - 1;
+      var longEnough = makeDistArray(size, bool);
+      if proper {
+        longEnough = (lengths > n);
+      } else {
+        longEnough = (lengths >= n);
+      }
+      var nFound = + reduce longEnough;
+      var retOffsets = makeDistArray(nFound, int);
+      forall (i, o) in zip(retOffsets.domain, retOffsets) {
+        o = i * (n + 1);
+      }
+      var retBytes = makeDistArray(nFound * (n + 1), uint(8));
+      var srcInds = makeDistArray(nFound * (n + 1), int);
+      var dstInds = (+ scan longEnough) - longEnough;
+      ref oa = offsets.a;
+      if kind == Fixes.prefixes {
+        forall (d, o, le) in zip(dstInds, oa, longEnough) with (var agg = newDstAggregator(int)) {
+          if le {
+            const srcIndStart = d * (n + 1);
+            for j in 0..#(n+1) {
+              agg.copy(srcInds[srcIndStart + j], o + j);
+            }
+          }
+        }
+      } else if kind == Fixes.suffixes {
+        forall (d, o, l, le) in zip(dstInds, oa, lengths, longEnough) with (var agg = newDstAggregator(int)) {
+          if le {
+            const srcIndStart = d * (n + 1);
+            const byteStart = o + l - n;
+            for j in 0..#(n+1) {
+              agg.copy(srcInds[srcIndStart + j], byteStart + j);
+            }
+          }
+        }
+      }
+      ref va = values.a;
+      forall (b, i) in zip(retBytes, srcInds) with (var agg = newSrcAggregator(uint(8))) {
+        agg.copy(b, va[i]);
+      }
+      return (retOffsets, retBytes, longEnough);
+    }
+
+    proc getFixes(n: int, kind: Fixes, proper: bool, param returnOrigins) where returnOrigins {
+      return getFixesHelper(n, kind, proper);
+    }
+
+    proc getFixes(n: int, kind: Fixes, proper: bool, param returnOrigins) where !returnOrigins {
+      var (off, byt, le) = getFixesHelper(n, kind, proper);
+      return (off, byt);
+    }
```

Comment:
if we go with the delayed action on `returnOrigins`, we should need to do this anymore since we can just call `getFixesHelper` directly

Reply:
done!